### DISCUSSION TITLE
How to use a board specific linker script without modifying platform.txt

### DISCUSSION THREAD

**Discussion by MaxLec14** [earlephilhower/arduino-pico#2307] [Q&A] [answered]
I'm using a custom variant of rp2040 and I want to have project specific custom linker scripts for it.  Right now im modifying the memmap_default in the lib folder.  I see that boards have a line  myboard.build.ldscript=memmap_default.ld but I cant seem to find where it's taken into account in the build process.  When I look at platform.txt all I see is that its hardcoded to the one in /lib.  Is it possible and am I wrong?

**Comment by earlephilhower** [comment, marked as answer]
I was planning on enabling it through that boards.txt entry.  But, since every RP2040 board has the same memmap and no external SRAM/etc. that the linker needs to worry about, I never got around to it.  Right now editing `platform.txt` like you're doing is the only way.  It should be cleaned up eventually, I suppose...

Note that OTA, core, filesystems/EEPROM, and final link stages assume certain things about the layout and if you drop things or rearrange them it might not work too well.
